.github/workflows/main.yml

```diff
@@ -54,9 +54,47 @@ jobs:
         env:
           ENABLE_SEARCH: 'true'
 
+      - name: Initialize Deployment
+        uses: bobheadxi/deployments@v0.6.2
+        id: start-deployment
+        with:
+          step: start
+          token: ${{ secrets.GITHUB_TOKEN }}
+          env: ${{ github.ref_name }}
+
+      - name: Deploy to Vercel
+        id: deploy-to-vercel
+        uses: amondnet/vercel-action@v20
+        with:
+          vercel-project-name: commercetools-docs-kit
+          vercel-token: ${{ secrets.VERCEL_TOKEN }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          # vercel-args: '--prod'  # TODO only of on main branch
+          vercel-org-id: ${{ secrets.VERCEL_ORG_ID}}
+          scope: commercetools
+          vercel-project-id: ${{ secrets.VERCEL_PROJECT_ID}}
+          # there are length restrictions to domains and only *.commercetools.vercel.app works
+          # reliably for generated subdomains. No branch name domain because branch names can be long.
+          alias-domains: |
+            docskit-sha-${{ github.sha }}.commercetools.vercel.app
+            docskit-pr-{{PR_NUMBER}}.commercetools.vercel.app
+
+      - name: Update Deployment Status
+        uses: bobheadxi/deployments@v0.6.2
+        id: finish_deployment
+        with:
+          step: finish
+          token: ${{ secrets.GITHUB_TOKEN }}
+          status: ${{ job.status }}
+          env_url: ${{ steps.deploy-to-vercel.outputs.preview-url }}
+          deployment_id: ${{ steps.start-deployment.outputs.deployment_id }}
+
+      # TODO move this into a separate workflow that holds the static checks only? They can run independently of builds
       - name: Lint text files
         run: yarn run jest --projects jest.{eslint,stylelint,test,text}.config.js --maxWorkers=3 --reporters jest-silent-reporter
```

Comment:
I would still run this before we deploy, to avoid unnecessary deployments if something does not work.

Reply:
OK, makes sense for the kit.  In the content repos the static checks on content are independent of the deployment so you always have  preview even if there are still formal issues.